src/vs/workbench/electron-browser/shell.ts

```diff
@@ -508,13 +512,77 @@ export class WorkbenchShell extends Disposable {
 		// Keep font info for next startup around
 		saveFontInfo(this.storageService);
 
+		this._savePartsSplash();
+
 		// Dispose Workbench
 		if (this.workbench) {
 			this.workbench.dispose(reason);
 		}
 	}
+
+	private _savePartsSplash() {
+
+		// capture html-structure
+		let html = '<div id="monaco-parts-splash">';
+
+		// title part
+		let titleHeight: number;
+		{
+			let part = this.workbench.getContainer(Parts.TITLEBAR_PART);
+			let pos = getDomNodePagePosition(part);
+			let bg = part.style.backgroundColor || 'inhert';
+			html += `<div style="position: absolute; width: 100%; left: 0; top: 0; height: ${pos.height}px; background-color: ${bg};"></div>`;
+			titleHeight = pos.height;
+		}
+
+		// activitybar-part
+		let left = this.workbench.getSideBarPosition() === Position.LEFT;
+		let activityPartWidth: number;
+		{
+			let part = this.workbench.getContainer(Parts.ACTIVITYBAR_PART);
+			let pos = getDomNodePagePosition(part);
+			let bg = part.style.backgroundColor || 'inhert';
+			html += `<div style="position: absolute; height: calc(100% - ${titleHeight}px); top: ${titleHeight}px; ${left ? 'left' : 'right'}: 0; width: ${pos.width}px; background-color: ${bg};"></div>`;
+			activityPartWidth = pos.width;
+		}
+
+		// sidebar-part
+		{
+			let part = this.workbench.getContainer(Parts.SIDEBAR_PART);
+			let pos = getDomNodePagePosition(part);
+			let bg = part.style.backgroundColor || 'inhert';
+			html += `<div style="position: absolute; height: calc(100% - ${titleHeight}px); top: ${titleHeight}px; ${left ? 'left' : 'right'}: ${activityPartWidth}px; width: ${pos.width}px; background-color: ${bg};"></div>`;
+		}
+
+		// statusbar-part
+		{
+			let part = this.workbench.getContainer(Parts.STATUSBAR_PART);
+			let pos = getDomNodePagePosition(part);
+			let bg = part.style.backgroundColor || 'inhert';
+
+			html += `<div style="position: absolute; width: 100%; bottom: 0; left: 0; height: ${pos.height}px; background-color: ${bg};"></div>`;
+		}
+
+		html += '\n</div>';
+
+		// store per workspace or globally
+		let state = this.contextService.getWorkbenchState();
+		if (state === WorkbenchState.EMPTY) {
+			this.storageService.store('parts-splash', html, StorageScope.GLOBAL);
+		} else {
+			this.storageService.store('parts-splash', html, StorageScope.WORKSPACE);
+		}
+	}
+
+	private _removePartsSplash(): void {
+		let element = document.getElementById('monaco-parts-splash');
```

Comment:
Can we put this `monaco-parts-splash` into a variable since it is used twice

Reply:
https://github.com/Microsoft/vscode/pull/53871/commits/25f44fb0f21f2fdd0899e90660aeacd00f69f69e